### PR TITLE
AWS Lambda Runtime legacy internal handlers need to be ignored from being instrumented and so traced …

### LINKED ISSUE #10931: Double Server span create by lambda instrumentation
### Describe the bug

This relates to this [issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7808). The bug still exists when running on Java 8 or 11 lambda runtimes. One thing that's different is the customer class loader is `lambdainternal.CustomerClassLoader` which is different from `com.amazonaws.services.lambda.runtime.api.client.CustomerClassLoader`

### Steps to reproduce

Deploy and invoke a sqs lambda handler with the javaagent. Ensure the handler name has `::handleRequest` suffix and not just the class name.

### Expected behavior

Only one server span.

### Actual behavior

Two server spans were created

### Javaagent or library instrumentation version

2.2.0

### Environment

**JDK**: 8 and 11
**OS**: linux

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by serkan-ozal**
... Otherwise, there might be double root spans.

Fixes #10931 

In Java 8 and Java 11 runtimes, AWS Lambda runtime is packaged under `lambdainternal` package (it is `com.amazonaws.services.lambda.runtime.api.client` for new runtime likes Java 17 and Java 21)

I have 
- zipped `/var/runtime/` folder in the AWS Lambda environment for Java 8 and Java 11 runtimes, 
- uploaded it to S3, 
- downloaded it to my local 
- and then unzipped

Here are the existing files:
Screenshot 2024-03-23 at 17 01 27

Then I have decompiled classes from `LambdaSandboxJava-byol.jar` and `LambdaJavaRTEntry-byol.jar`.

And as you can see below, AWS Lambda runtime internal classes are packaged under `lambdainternal` package and the internal handler (`lambdainternal.EventHandlerLoader`) we need to ignore is there.

Screenshot 2024-03-23 at 16 41 03

### CODE REVIEW

**Review by laurit — APPROVED** [review]

**Review by tylerbenson — APPROVED** [review]
Wow.  Love the deep analysis you went through for this.  That seems worthy of a blog post somewhere.  Thanks for getting to the bottom of this.

File: instrumentation/aws-lambda/aws-lambda-core-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awslambdacore/v1_0/AwsLambdaRequestHandlerInstrumentation.java [hunk 1]
```diff
@@ -38,7 +38,8 @@ public ElementMatcher<ClassLoader> classLoaderOptimization() {
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
     return implementsInterface(named("com.amazonaws.services.lambda.runtime.RequestHandler"))
-        .and(not(nameStartsWith("com.amazonaws.services.lambda.runtime.api.client")));
+        .and(not(nameStartsWith("com.amazonaws.services.lambda.runtime.api.client")))
+        .and(not(nameStartsWith("lambdainternal")));
```
Comment (trask) [inline instrumentation/aws-lambda/aws-lambda-core-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awslambdacore/v1_0/AwsLambdaRequestHandlerInstrumentation.java:42]:
```suggestion
        // In Java 8 and Java 11 runtimes, AWS Lambda runtime is packaged under `lambdainternal` package
        // (it is `com.amazonaws.services.lambda.runtime.api.client` for new runtime likes Java 17 and Java 21)
        .and(not(nameStartsWith("lambdainternal")));
```

**Review by trask — APPROVED** [review]
Thanks!

**Comment by serkan-ozal** [comment]
Hi @trask,

Thanks for the review, I have applied your suggestion.

**Pull request merged**